test/dbus.c

```diff
@@ -660,6 +660,34 @@ TEST test_hint_raw_image(void)
         PASS();
 }
 
+/* We didn't process the timeout parameter via DBus correctly
+ * and it got limited to an int instead of a long int
+ * See: Issue #646 (The timeout value in dunst wraps around) */
+TEST test_timeout_overflow(void)
+{
+        struct notification *n;
+        struct dbus_notification *n_dbus;
+
+        n_dbus = dbus_notification_new();
+        n_dbus->app_name = "dunstteststack";
+        n_dbus->app_icon = "NONE";
+        n_dbus->summary = "test_hint_urgency";
+        n_dbus->body = "Summary of it";
+        n_dbus->expire_timeout = 2147484;
+        long int expected_timeout = 2147484000ll;
```

Comment:
Nitpick since it doesn't really affect anything in this case but: You're assigning a `long long int` to a `long int`.

Reply:
So, I've used the GLib way.

Idk why GLib is only using `long` as their 64 bit specifier. According to the german wikipedia-article this is not true in all cases, but it seems to work and the GLib guys have a reason to use only one `l`.